src/legacy/core_plugins/vis_default_editor/public/components/controls/time_interval.tsx

```diff
@@ -118,6 +120,11 @@ function TimeIntervalParamEditor({
     setValidity(isValid);
   }, [isValid]);
 
+  // To update aggConfigs actual timeRange, value of timeRange is set from context.
+  useEffect(() => {
+    agg.aggConfigs.setTimeRange(timeRange);
+  }, [agg.aggConfigs, timeRange]);
+
```

Comment:
If we actually start calling `setTimeRange` on `AggConfigs` again (which I am btw appreciating), I think we should do that in the "higher" component in the tree, i.e. in this case the `agg.tsx` component. I suggest we move that logic in the hook we have in that component.

Since we'll have the issue, that this component then somehow also needs to rerender, when the timeRange changes, I'd suggest, we actually store the `interval` above in a state variable (since this is the variable actually affected by changes in the `timeRange`). Then we just have an effect hook, here that updates the `interval` state variable, and thus triggers a rerender. And we'd then use that `interval` from state, instead of a local variable as above, e.g. (untested) example code:

```ts


const timeRange = useContext(DefaultEditorAggParamsContext);
const getBucketInterval = () => get(agg, 'buckets.getInterval') && (agg as any).buckets.getInterval();
const [interval, setInterval] = useState(getBucketInterval());

useEffect(() => {
  setInterval(getBucketInterval());
}, [timeRange]);

// .. the remaining code as it is, consuming `interval`
```

Reply:
Turned out that the whole component DefaultEditorAgg is rerendered when its state is updated, so there's no need in this useEffect. Please, check the updated description.

Reply:
Let's cover this functionality with a set of functional tests (check accordion label changing, check interval param help text changing), to make sure we'll not miss it anymore.